opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -713,6 +714,30 @@ std::vector<std::array<int,2>> CpGrid::mapLeafIndexSetToLocalCartesianIndexSets(
     return leafIdx_to_localCartesianIdxSets;
 }
 
+std::vector<std::array<int,3>> CpGrid::localIJK(const std::string& lgr_name) const
+{
+    // Check lgr_name exists in lgr_names_
+    auto it = lgr_names_.find(lgr_name);
+    if (it == lgr_names_.end()) {
+        OPM_THROW(std::runtime_error, "LGR name not found: " + lgr_name);
+    }
+
+    const Opm::LevelCartesianIndexMapper<Dune::CpGrid> levelCartMapp(*this);
+    const auto& level = it->second;
+
+    std::vector<std::array<int, 3>> localIJK;
+    localIJK.reserve(levelGridView(level).size(0));
+
+    for (const auto& element : elements(levelGridView(level))) {
+        std::array<int, 3> local_ijk;
+        levelCartMapp.cartesianCoordinate(element.getEquivLevelElem().index(), local_ijk, element.level());
```

Comment:
This looks a bit to complicated. `element` should be on the level and should be passed directly.

Why do we need to specify the last parameter? Shouldn't the level should be implied by the element.

Reply:
Indeed, it's enough with element.index(). The last parameter we need since the LevelCartesianIndexMapper takes a CpGrid in it's constructor, instead of a CpGridaData. Then, you should specify the level. 

If you see any improvements on LevelCartesianIndexMapper, we can change that too. It might not be the best, but it does what it was created for, namely, the transaltion between indices and the local/level Cartesian indices of the underlying Cartesian grid associated with the LGR